src/views/view.c

```diff
@@ -850,7 +850,9 @@ const GList *dt_view_get_images_to_act_on(const gboolean only_visible, const gbo
           // selection already contains them :)
           DT_DEBUG_SQLITE3_PREPARE_V2
             (dt_database_get(darktable.db),
-             "SELECT DISTINCT imgid FROM main.selected_images", -1, &stmt, NULL);
+             "SELECT DISTINCT imgid"
+             " FROM main.selected_images"
+             " ORDER BY imgid DESC", -1, &stmt, NULL);
```

Comment:
This force to order by imgid, but anyway otherwise the order is undefined. So better do to it like that :)